test/extended/baremetal/hosts.go

```diff
@@ -204,14 +109,53 @@ var _ = g.Describe("[sig-installer][Feature:baremetal] Baremetal platform should
 		g.By("updating bootMACAddress which is not allowed")
 		err = unstructured.SetNestedField(host.Object, testMACAddress, "spec", "bootMACAddress")
 		o.Expect(err).NotTo(o.HaveOccurred())
-		_, err = bmc.Update(context.Background(), &host, v1.UpdateOptions{})
+		_, err = bmc.Update(context.Background(), &host, metav1.UpdateOptions{})
 		o.Expect(err).To(o.HaveOccurred())
 		o.Expect(err.Error()).To(o.ContainSubstring("bootMACAddress can not be changed once it is set"))
 
 		g.By("verify bootMACAddress is not updated")
-		h, err := bmc.Get(context.Background(), host.GetName(), v1.GetOptions{})
+		h, err := bmc.Get(context.Background(), host.GetName(), metav1.GetOptions{})
 		o.Expect(err).NotTo(o.HaveOccurred())
 		check, _, _ := unstructured.NestedString(h.Object, "spec", "bootMACAddress")
 		o.Expect(check).To(o.Equal(bootMACAddress))
 	})
 })
+
+// This block must be used for the serial tests. Any eventual extra worker deployed will be
+// automatically deleted during the AfterEach
+var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] Baremetal platform should", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc     = exutil.NewCLI("baremetal")
+		helper *BaremetalTestHelper
+	)
+
+	g.BeforeEach(func() {
+		helper = NewBaremetalTestHelper(oc)
+		helper.Setup()
+	})
+
+	g.AfterEach(func() {
+		helper.DeleteAllExtraWorkers()
+	})
+
+	g.It("skip inspection when disabled by annotation", func() {
+		skipIfNotBaremetal(oc)
+
+		// Get extra worker info
+		hostData, secretData := helper.GetExtraWorkerData(0)
+
+		// Set inspection annotation as disabled
+		unstructured.SetNestedField(hostData.Object, "disabled", "metadata", "annotations", "inspect.metal3.io")
+
+		// Deploy extra worker and wait
+		host, _ := helper.CreateExtraWorker(hostData, secretData)
+		host = helper.WaitForProvisioningState(host, "available")
+
+		g.By("Check hardware to be empty")
+		_, found, err := unstructured.NestedString(host.Object, "status", "hardware")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(found).To(o.BeFalse())
```

Comment:
So hardware is a pointer to a HardwareDetails struct, shouldn't there be a check for the value being nil instead of it being found?

Reply:
We don't have a structured type, so if not present it does not get unmarshalled at all (and thus the need to check the bool flag)